Update docs to clarify the way aliases are used in CTEs (#5795)
fishtown-analytics/docs.getdbt.com: website/docs/docs/build/custom-aliases.md

## Overview

> When dbt runs a model, it will generally create a relation (either a <Term id="table" /> or a <Term id="view" />) in the database, except in the case of an [ephemeral model](/docs/build/materializations), when it will create a <Term id="cte" /> for use in another model. By default, dbt uses the model's filename as the identifier for the relation or CTE it creates. This identifier can be overridden using the [`alias`](/reference/resource-configs/alias) model configuration.

Reviewer: ```suggestion
When dbt runs a model, it will generally create a relation (either a <Term id="table" /> or a <Term id="view" /> ) in the database, except in the case of an [ephemeral model](/docs/build/materializations), when it will create a <Term id="cte" /> for use in another model. By default, dbt uses the model's filename as the identifier for the relation or CTE it creates. This identifier can be overridden using the [`alias`](/reference/resource-configs/alias) model configuration.
```

> When dbt runs a model, it will generally create a relation (either a <Term id="table" /> or a <Term id="view" />) in the database, except in the case of an [ephemeral model](/docs/build/materializations#ephemeral), when it will create a <Term id="cte" /> for use in another model. By default, dbt uses the model's filename as the identifier for the relation or CTE it creates. This identifier can be overridden using the [`alias`](/reference/resource-configs/alias) model configuration.

Reviewer: ```suggestion
When dbt runs a model, it will generally create a relation (either a <Term id="table" /> or a <Term id="view" />) in the database, except in the case of an [ephemeral model](/docs/build/materializations), when it will create a <Term id="cte" /> for use in another model. By default, dbt uses the model's filename as the identifier for the relation or CTE it creates. This identifier can be overridden using the [`alias`](/reference/resource-configs/alias) model configuration.
```

> When dbt runs a model, it will generally create a relation (either a <Term id="table" /> or a <Term id="view" />) in the database, or in the case of an [ephemeral model](/docs/build/materializations#ephemeral) it will create a <Term id="cte" /> for use in another model. By default, dbt uses the filename of the model as the identifier for the relation or CTE it creates. This identifier can optionally be overridden using the [`alias`](/reference/resource-configs/alias) model configuration.

Reviewer: ```suggestion
When dbt runs a model, it will generally create a relation (either a <Term id="table" /> or a <Term id="view" />) in the database, except in the case of an [ephemeral model](/docs/build/materializations#ephemeral), when it will create a <Term id="cte" /> for use in another model. By default, dbt uses the model's filename as the identifier for the relation or CTE it creates. This identifier can be overridden using the [`alias`](/reference/resource-configs/alias) model configuration.
```

When dbt runs a model, it will generally create a relation (either a <Term id="table" /> or a <Term id="view" />) in the database, or in the case of an [ephemeral model](/docs/build/materializations#ephemeral) it will create a <Term id="cte" /> for use in another model. By default, dbt uses the filename of the model as the identifier for the relation or CTE it creates. This identifier can optionally be overridden using the [`alias`](/reference/resource-configs/alias) model configuration.

### Why alias model names?
The names of schemas and tables are effectively the "user interface" of your <Term id="data-warehouse" />. Well-named schemas and tables can help provide clarity and direction for consumers of this data. In combination with [custom schemas](/docs/build/custom-schemas), model aliasing is a powerful mechanism for designing your warehouse.

### Usage

> The `alias` config can be used to change the name of a model's identifier in the database. The following <Term id="table" /> shows examples of database identifiers for models both with, and without, a supplied `alias`.

Author: I tweaked this line to remove the misleading `<Term>` component, since in this case we're not actually talking about a relation (which is what the term documents), we're just referencing the markup table that follows.

Reviewer: Good call! Thank you

The `alias` config can be used to change the name of a model's identifier in the database. The following table shows examples of database identifiers for models both with and without a supplied `alias`, and with different materializations.

| Model | Config | Relation Type | Database Identifier |
| ----- | ------ | --------------| ------------------- |
| ga_sessions.sql | {{ config(materialization='view') }} | <Term id="view" /> | "analytics"."ga_sessions" |
| ga_sessions.sql | {{ config(materialization='view', alias='sessions') }} | <Term id="view" /> | "analytics"."sessions" |
| ga_sessions.sql | {{ config(materialization='ephemeral') }} | <Term id="cte" /> | "\__dbt\__cte\__ga_sessions" |

> | ga_sessions.sql | {{ config(materialization='ephemeral', alias='sessions') }} | <Term id="cte" /> | "\__dbt\__cte\__sessions" |

Author: I'm not actually sure whether it's helpful to document the CTE identifier behavior here, so I understand if you'd prefer to keep this table focused on the alias rather than explaining the different behavior of ephemeral materialization.

Reviewer: Very helpful! Thank you
